core/src/main/java/io/smallrye/context/impl/cdi/SmallRyeCurrentThreadContextInterceptor.java

```diff
@@ -33,15 +35,25 @@ public Object manageCurrentContext(InvocationContext ctx) throws Exception {
         if (config == null && ctx.getMethod() != null)
             config = ctx.getMethod().getAnnotation(CurrentThreadContext.class);
         if (config != null) {
-            SmallRyeThreadContext newTC = config.remove()
-                    ? null
-                    : SmallRyeThreadContext.builder()
-                            .cleared(config.cleared())
-                            .propagated(config.propagated())
-                            .unchanged(config.unchanged())
-                            .build();
-            try (CleanAutoCloseable ac = SmallRyeThreadContext.withThreadContext(newTC)) {
+            final String[] unchanged = config.unchanged();
+            final String[] propagated = config.propagated();
+            final String[] cleared = config.cleared();
+            if (propagated.length == 0 &&
+                    cleared.length == 0 &&
+                    unchanged.length == 1 && ThreadContext.ALL_REMAINING.equals(unchanged[0])) {
```

Comment:
I feel like we should be able to skip this last check, because the config has been validated, and if someone crazy lists all the unchanged contexts one by one it will be equivalent. The important part is that nothing is propagated and nothing is cleared.

Reply:
nice :) I'll amend

Reply:
done